common/SmokeTests/SmokeTest/Program.cs

```diff
@@ -1,37 +1,227 @@
+// enumerated, and instances will be created for any type with a parameterless constructor.
+//
+// This will ensure that all referenced packages have been loaded and can be used in
+// the same application without requiring specific knowledge of each SDK or code related to
+// its functionality.
+//
+// Execution will be logged to the console, using standard output for informational messages
+// and standard error for errors.   A return code of "0" will indicate a successful run; a
+// non-zero return indicates a test failure.
+
+const string AssemblyFileMask = "*.dll";
+
+LogInformation($"Smoke Test run starting...{ Environment.NewLine }");
+
+var returnCode = 0;
+
+// Starting with the entry assembly as the root, walk all references, both
+// direct and transient to process them.
+
+foreach (var assembly in LoadAssemblies(Assembly.GetEntryAssembly(), AssemblyFileMask))
 {
-    class Program
+    LogInformation($"Processing assembly: [{ assembly.FullName }]...");
+
+    var typeCount = 0;
+
+    try
     {
-        static async Task<int> Main(string[] args)
+        // Process each type in the assembly.  The call to "ProcessType" is safe and
+        // will not throw.
+
+        foreach (var type in assembly.GetTypes())
         {
-            Console.WriteLine("SMOKE TEST FOR TRACK 2 LIBRARIES");
-
-            await KeyVaultTest.RunTests();
-            await BlobStorageTest.RunTests();
-            await EventHubsTest.RunTests();
-            // Temporarily disable Cosmos DB smoke tests
-            // Issue to re-enable:
-            // https://github.com/Azure/azure-sdk-for-net/issues/11297
-            // await CosmosDBTest.RunTests();
-
-            Console.WriteLine();
-            Console.WriteLine("LIBRARY TESTS COMPLETE");
-            Console.WriteLine();
-            Console.WriteLine("SMOKE TEST FOR TRACK 2 SAMPLES");
-
-            await IotHubConnectionTests.RunTests();
-
-            Console.WriteLine("SAMPLES TESTS COMPLETE");
-            Console.WriteLine();
-            return 0;
+            ++typeCount;
+            ProcessType(type);
         }
     }
+    catch (Exception ex)
+    {
+        ReportError();
+        LogError($"{Environment.NewLine }[{ assembly.FullName }] could not be fully processed.  Error: { ex }{ Environment.NewLine }");
+    }
+
+    LogInformation($"Completed assembly: [{ assembly.FullName }].  { typeCount } types inspected.{ Environment.NewLine }");
+}
+
+// End the test run and report on the state.
+
+LogInformation($"{ Environment.NewLine }Smoke Test run complete.");
+
+if (returnCode == 0)
+{
+    LogInformation("Passed.", ConsoleColor.Green);
+}
+else
+{
+    LogInformation("Failed.", ConsoleColor.Red);
 }
+
+return returnCode;
+
+// Local function definitions
+
+void ReportError() => returnCode = -1;
+
+void LogError(string message, ConsoleColor? color = default)
+{
+    Console.ForegroundColor = color ?? ConsoleColor.Red;
+    Console.Error.WriteLine(message);
+    Console.ResetColor();
```

Comment:
Tip: it's unlikely, but if anything throws this won't work. Instead, I like to use this pattern:

```csharp
var previousForeground = Console.ForegroundColor;
Console.ForegroundColor = color ?? ConsoleColor.Red;
try
{
  // Do whatever
}
finally
{
  Console.ForegroundColor = previousForeground;
}
```

Granted, I wouldn't expect just writing to the console to fail, but if this ever does anything more - like maybe formatting an event, exception, etc. - it's more likely.

Reply:
I don't see this doing anything additional, but the extra safety won't hurt.    Updated patterns in the main PR.